lib/parsers/yaml.js

```diff
@@ -37,19 +30,20 @@ module.exports = {
    * @returns {Promise}
    */
   parse (file) {
-    return new Promise(((resolve, reject) => {
-      let data = file.data;
-      if (Buffer.isBuffer(data)) {
-        data = data.toString();
-      }
+    let data = file.data;
+    if (Buffer.isBuffer(data)) {
+      data = data.toString();
+    }
 
-      if (typeof data === "string") {
-        resolve(YAML.parse(data));
-      }
-      else {
-        // data is already a JavaScript value (object, array, number, null, NaN, etc.)
-        resolve(data);
-      }
-    }));
+    if (typeof data === "string") {
+      return YAML.parse(data);
```

Comment:
Should this return the `{data, errors}` object? If so, should we add a test to cover this code branch?

Reply:
It's done in the `yaml` implementation.
https://github.com/stoplightio/json-schema-ref-parser/pull/1/files#diff-21e5cca401153ae38576cef3ccfbb990R26